File: 020922/Code/USACO_21Dec_Bronze_1.cpp
#include<bits/stdc++.h>
using namespace std;
typedef long long ll;

int n;
char s[505050];
int a[505050];
int cnt;
ll res;
	
int main(){
	scanf("%d", &n);
	scanf("%s", s);
	
	cnt = 0;
	for (int i = 0; i < n; i++){
		if (i == 0 || s[i] != s[i - 1]) cnt++;
		a[cnt]++;
	}
	
	res = 0;
	for (int i = 1; i < cnt; i++){
		res += max(0, a[i] - 1);
	}
	for (int i = 2; i <= cnt; i++){
		res += max(0, a[i] - 1);
	}
	for (int i = 2; i < cnt; i++){
		if (a[i] == 1) res += 1ll * a[i - 1] * a[i + 1];
	}
	printf("%lld\n", res);
	return 0;
}

File: 020922/Code/USACO_21Dec_Bronze_1.py
n = int(input())
s = input()

cnt = 0
tmp = 0
a = []

for i in range(len(s)):
    if i > 0 and s[i] != s[i - 1]:
        cnt = cnt + 1
        a.append(tmp)
        tmp = 0
    tmp = tmp + 1
a.append(tmp)
cnt = cnt + 1
res = 0
for i in range(cnt):
    if i > 0:
        res = res + max(0, a[i] - 1)
    if i < cnt - 1:
        res = res + max(0, a[i] - 1)
    if 0 < i < cnt - 1 and a[i] == 1:
        print("a[{}] = {}".format(i, a[i]))
        res = res + a[i - 1] * a[i + 1]



print(res)